jax/experimental/jax2tf/impl_no_xla.py

```diff
@@ -792,10 +792,99 @@ def _dynamic_update_slice(operand, update, *start_indices,
 
 tf_impl_no_xla[lax.dynamic_update_slice_p] = _dynamic_update_slice
 
-tf_impl_no_xla[lax.scatter_p] = _unimplemented("scatter")
-tf_impl_no_xla[lax.scatter_min_p] = _unimplemented("scatter min")
-tf_impl_no_xla[lax.scatter_max_p] = _unimplemented("scatter max")
-tf_impl_no_xla[lax.scatter_mul_p] = _unimplemented("scatter mul")
-tf_impl_no_xla[lax.scatter_add_p] = _unimplemented("scatter add")
+def get_move_to_leading_axis_bijector(axis):
+  '''
+  Bijector taking an array x, of rank `operand_rank`.
+  Forward shift the axis `axis` to the leading axis
+
+  so for array `operand_rank`=4 and `axis=4`. The last axis is
+  moved for the front: permutation - [3,0,1,2]
+  '''
+  def shift(operand, inverse=False):
+    operand_rank = tf.rank(operand)
+    fwd_order = tf.concat([[axis], tf.range(axis), tf.range(axis+1, operand_rank)], 0)
+    order = fwd_order if not inverse else tf.math.invert_permutation(fwd_order)
+    return tf.transpose(operand, order)
+  _forward = shift
+  _inverse = partial(shift, inverse=True)
+  return _forward, _inverse
+
+def convert_scatter_jax_to_tf(update_op):
+  def error(msg):
+    suffix = ("See source code for the precise conditions under which "
+              "scatter_(update/add/multiply/min/max) ops can be converted without XLA.")
+    return _xla_disabled_error("scatter_(update/add/multiply/min/max)", f"{msg} - {suffix}")
+
+  def _sparse_scatter(
+    operand,
+    scatter_indices,
+    updates,
+    update_jaxpr,
+    update_consts,
+    dimension_numbers,
+    indices_are_sorted: bool,
+    unique_indices: bool,
+    mode,
+    _in_avals: Sequence[core.ShapedArray],
+    _out_aval: core.ShapedArray):
+    '''
+    Implementation of sparse scatter that works for cases when
+    operand (N dimensional tensor) is indexed in the leading dimension by 1D array `scatter_indices` with updates, an (N-1) dimensional tensor
+    '''
+    suboperand = tf.gather_nd(operand, scatter_indices)
+    updated_suboperand = update_op(suboperand, updates)
+    return tf.tensor_scatter_nd_update(operand, scatter_indices[None], updated_suboperand[None])
```

Comment:
Perhaps it is useful to explain in a comment why you need to add an empty dimension to `scatter_indices` and `updated_suboperand`.

Reply:
Altered doc string. It comes down to:
XLAGather has dimension numbers which deal with specifying batch dimensions and which dims the scatters go in to. tf.tensor_scatter_nd_update does not have dimension numbers so has to infer everything from the shapes, it indexs from the front only and forcibly has at least one batch dimension.

For example, If you have 4D array, but are indexing into the last 2 dims e.g. x[:,:,0,0] you need to shift the last two dims to the front to use a tf equivalent, you can't just 'pass' on a dim.